setdate.js

```diff
@@ -0,0 +1,87 @@
+
+var myDate = (function(){
+
+  var __date = null;
+  var days = ["Sunday", "Monday", "Tuesday", "Wednesday",
+                "Thursday", "Friday", "Saturday"];
+  var months = ["January", "Febuary", "March", "April", "May",
+                  "June", "July", "August", "September",
+                  "October", "November", "December"];
+
+
+  function __checkIfInstantiated(){
+    if (__date) { return __date; } else { return new Date(); };
+  };
+
+  function __milliseconds(){
+    var date = __checkIfInstantiated();
+
+    return date.getTime();
+  };
+
+  function __formatted(){
+    var date = __checkIfInstantiated();
+    var month = getMonthName();
+    var day = date.getDate();
+    var year = date.getFullYear();
+
+    return month + " " + day + ", " + year;
+  };
+
+  function setDate(value){
+    value = value || Date.now();
+    __date = new Date(value);
```

Comment:
Ideally this method would do a little more type-checking. If value is an integer or a date instance, then you'd want to instantiate your date using value. Currently it's possible to pass in invalid arguments and the method will try to instantiate a new Date instance using it.